### ISSUE TITLE
travis: segfault in lua/t1002-kvs.t

### ISSUE THREAD

**Issue by grondo** [flux-framework/flux-core#863]
After the merge of #862, the GCC build in travis failed with a segfault in one of the Lua tests:

```
2016-10-21T05:27:02.306801Z broker.err[0]: Run level 2 Segmentation fault (rc=139) 1.8s
flux-start: 0 (pid 105413) exited with rc=139 (Segmentation fault)
FAIL: lua/t1002-kvs.t
```

This is reproducible after a restart, but I can't reproduce it locally, so perhaps some combination of libraries/compiler version in that specific travis build, cached files, etc. It also didn't occur on the PR test branch so there is something specific to this build.

Unfortunately the backtrace-all script failed to get a backtrace with a "cannot find executable" error, so there is no good information about what happened.

I'm going to clear the master branch caches and rebuild, to see if that clears the error.

**Comment by garlick** [comment]
Uh oh.  Do you read the above as a segfault in the initial program (I guess the lua interpreter in this case?)  If so then I wonder if something bad happened in #857 (merged Thurs afternoon) which, unlike #862, does touch the lua bindings...

**Comment by grondo** [comment]
Yeah, segfault was in the Lua interpreter, and after all tests were complete (so perhaps in final garbage collection). However, only the GCC builder saw this issue, and I just cleared the cache and rebuilt and the problem didn't recur, so I'm willing to guess some kind of environment or state issue.

I also found a bug in the `backtrace-all.sh` script due to this, so I'll update that script and hopefully we'll get more information if this occurs again. (Meanwhile I'll also run this test under valgrind just to be safe)
